components/pluralrules/src/operands.rs

```diff
@@ -0,0 +1,275 @@
+use std::convert::TryFrom;
+use std::io::{Error as IOError, Write};
+use std::isize;
+use std::num::ParseIntError;
+use std::str::FromStr;
+
+/// A full plural operands representation of a number. See [CLDR Plural Rules](http://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules) for complete operands description.
+/// Plural operands in compliance with [CLDR Plural Rules](http://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules).
+///
+/// See [full operands description](http://unicode.org/reports/tr35/tr35-numbers.html#Operands).
+///
+/// # Examples
+///
+/// From int
+///
+/// ```
+/// use icu_pluralrules::PluralOperands;
+/// assert_eq!(PluralOperands {
+///    n: 2_f64,
+///    i: 2,
+///    v: 0,
+///    w: 0,
+///    f: 0,
+///    t: 0,
+/// }, PluralOperands::from(2_usize))
+/// ```
+///
+/// From float
+///
+/// ```
+/// use std::str::FromStr;
+/// use icu_pluralrules::PluralOperands;
+/// assert_eq!(Ok(PluralOperands {
+///    n: 1234.567_f64,
+///    i: 1234,
+///    v: 3,
+///    w: 3,
+///    f: 567,
+///    t: 567,
+/// }), "-1234.567".parse())
+/// ```
+///
+/// From &str
+///
+/// ```
+/// use std::convert::TryFrom;
+/// use icu_pluralrules::PluralOperands;
+/// assert_eq!(Ok(PluralOperands {
+///    n: 123.45_f64,
+///    i: 123,
+///    v: 2,
+///    w: 2,
+///    f: 45,
+///    t: 45,
+/// }), PluralOperands::try_from(123.45))
+/// ```
+#[derive(Debug, PartialEq)]
+pub struct PluralOperands {
+    /// Absolute value of input
+    pub n: f64,
+    /// Integer value of input
+    pub i: u64,
+    /// Number of visible fraction digits with trailing zeros
+    pub v: usize,
+    /// Number of visible fraction digits without trailing zeros
+    pub w: usize,
+    /// Visible fraction digits with trailing zeros
+    pub f: u64,
+    /// Visible fraction digits without trailing zeros
+    pub t: u64,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum OperandsError {
+    /// Input to the Operands parsing was empty.
+    Empty,
+    /// Input to the Operands parsing was invalid.
+    Invalid,
+}
+
+impl From<ParseIntError> for OperandsError {
+    fn from(_: ParseIntError) -> Self {
+        Self::Invalid
+    }
+}
+
+impl From<IOError> for OperandsError {
+    fn from(_: IOError) -> Self {
+        Self::Invalid
+    }
+}
+
+impl FromStr for PluralOperands {
```

Comment:
At some point, we are going to have to move this code into FixedDecimal (#180).

Since FixedDecimal is landed, why not do it now?  Why introduce code into the pluralrules crate that someone will have to come back and move into FixedDecimal.

Reply:
My main reason is to have performance evaluation and potential discussion about it in a separate PR from this one. There's a lot of perf optimizations here and I'd like to have each "swap" (fixed decimal and dataprovider namely) be done separately with full evaluation of the impact.

Reply:
Okay, I'm fine with that